iati/custom_middleware.py

```diff
@@ -27,7 +27,12 @@ def __call__(self, request):
         self.path_parts = self.remove_language_code(valid_path_parts)
         self.is_download = "downloads" in self.path_parts
         self.stripped_path = '/'.join(self.path_parts)
+        full_url = request.build_absolute_uri()
+        parsed_url = urllib.parse.urlparse(full_url)
+        domain = parsed_url.netloc
 
+        if domain == 'reference.iatistandard.org' and parsed_url.path == '':
```

Comment:
Testing this out with editing my /etc/hosts, and the parsed_url.path isn't blank:

```
web_1       | DOMAIN: reference.iatistandard.org
web_1       | PATH: /
```

Also to cover our bases, I think we should include the `/en/` and `/fr/` paths as some people may have cached redirects to those. And then lastly I wouldn't hardcode the /en/ in the redirect, and just redirect it to /iati-standard/. The locale redirects will direct the user to the correct /en/iati-standard or /fr/iati-standard based on their browser. So all together:

```
        if domain == 'reference.iatistandard.org' and parsed_url.path in ['/', '/en/', '/fr/']:
            return http.HttpResponsePermanentRedirect(f'{settings.REFERENCE_REDIRECT_BASE_URL}/iati-standard/')
```